lib/sidekiq/antidote/middlewares/shared.rb

```diff
@@ -17,7 +17,8 @@ def inhibit(message)
 
           Antidote.log(:warn) { "I've got a poison! -- #{job_record.display_class}" }
           Antidote.log(:warn) { "I've got a remedy! -- #{inhibitor}" }
-          DeadSet.new.kill(Sidekiq.dump_json(message)) if "kill" == inhibitor.treatment
+
+          inhibitor.apply(message)
```

Comment:
To ensure the message contains queue, let's do something like:

```suggestion
          inhibitor.apply(message.merge("queue" => queue))
```